### ISSUE TITLE
recursive invocation while running trap for signal 14 function

### ISSUE THREAD

**Issue by usuarionuevor** [FluxionNetwork/fluxion#1005]
launches AP, everything ok, but after a while it crash with the following:

.../fluxion/lib/IOUtils.sh: line 22: warning: run_pending_traps: recursive invocation while running trap for signal 14 function
Starting Captive Portal access point service...

.../fluxion/lib/ap/hostapd.sh: line 81: warning: run_pending_traps: recursive invocation while running trap for signal 14
.../fluxion/lib/ap/hostapd.sh: line 81: warning: run_pending_traps: recursive invocation while running trap for signal 14
and so on...

may be kali 2020.4 ? because its the first time I'm getting this error, over and over again

**Comment by usuarionuevor** [comment]
@strasharo why you closed? also happens with handshake snooper,
both works fine with airgeddon and kali 2020.4, do I need 2019.4 for fluxion?

**Comment by cjb900** [comment]
@usuarionuevor fluxion works on my end with the latest kali 2020.4 with all the updates. I would make sure you have all the dependencies for fluxion installed and working. you can even try reinstalling them and see if that helps. Also make sure none of the dependencies are being used by any other tool and or program as that could cause fluxion to not be able to use that dependence properly.

**Comment by usuarionuevor** [comment]
Thanks for answer, fluxion is up to date, and kali 2020.4 updated with `apt-get update`
I also tried running `airmon-ng check kill` before running fluxion but same result.

**Comment by usuarionuevor** [comment]
AP, deauth, everything works perfect at first, then the warnings shows after a while running and crash

**Comment by cjb900** [comment]
then my only guess is its either something corrupt or not installed properly in kali or one of the dependencies is not installed or working properly. I tested fluxion the other day and had no issues at all.

**Comment by cjb900** [comment]
also ```apt-get update``` is not how you check for updates in kali. All that does is check the repository kali uses is up to date it does not actually update anything in kali if there are updates/upgrades. I would run this command to make sure everything is up to date ```sudo apt update && sudo apt upgrade -y && sudo apt dist-upgrade -y```

**Comment by usuarionuevor** [comment]
ok, found the issue, after reinstalling a hundred of times I looked for another error I ignored before in AP Authenticator: "/tmp/fluxspace/clients.txt: No such file or directory"

and found similar (not the same) issue here: https://github.com/FluxionNetwork/fluxion/issues/539
^^last comment suggests to **SKIP selecting interface for target tracking** --> and this worked!

I tried with rt3070 (awus036nh), rtl8812au (awus036ach) and both fails, they are set OK, but after a while running it fails.

In kali 2018 it worked fine with any adapter, even with laptop default so I don't understand how it doesn't work now even with rt3070.

**Comment by cjb900** [comment]
Well the RT3070 chipset in the awus036nh is on known good chipset. I have an adapter with this chipset and it works well with fluxion in the latest version of kali with no issues. The rtl8812au chipset is a known one that does not properly work as RTL chipsets in general are notoriously problematic for wifi hacking. You can try running ``airmon-ng check kill`` before starting fluxion with the  awus036nh adapter and see if that helps any.

**Comment by usuarionuevor** [comment]
yes tried `airmon-ng check kill` but no difference, also I have to say I never used it before and never had a problem.

rtl8812au had bad drivers for a while, but isn't just target tracking to check if real AP is still in up and check the channel? I mean is not doing any injection or deauth or anything "complex" right?

how about Atheros QCA9565/AR9565 is the one by default on laptop and the one I always used for target tracking without any issue before, now is completely broken for target tracking? I don't get it
with the rtl8812au at least it restart AP after the "clients.txt no such file or directory" error so the AP keeps running for a while longer, is there any log I can provide?

**Comment by cjb900** [comment]
Well i'm not familiar with the Atheros QCA9565/AR9565. but if all you are using it for target tracking as long as it supports monitor mode it should work. one question, the AP you are attacking are you sure that it actually changes the channel its on when you deauth? because not all AP's do and none of the AP's around me change the channel so I don't use target tracking. If the AP does not change the channel its on when you deauth that may be part of the issue you are having as well.

**Comment by usuarionuevor** [comment]
yes it changes the channel, even if not attacking, modern boxes do this to choose the less saturated channel, can be configured to change channel every X hours, and also option to restart the AP at some time you specify (which also changes the channel).

what I observed the times it works, it gives the same errors, the no clients.txt in AP authenticator and warning with recursive invocation but then restart the attack with new AP window so it start over like if nothing happened, so my guess is the issue is sometimes get stuck at this point.

I noticed it restart the attack also when AP didn't change the channel, is this what you mean?

Also, when it works, if you run it for several hours since it keeps opening new AP windows to the infinite it tanks the laptop, can't this be done in different (more efficient) way?
